test/metabase/driver/postgres_test.clj

```diff
@@ -1396,3 +1415,18 @@
                (do-test :schema-pattern "private" :schemas ["private"]))
              (testing "filter by table name"
                (do-test :table-pattern "table" :tables ["table"])))))))))
+
+(deftest sync-estimated-row-count-test
+  (mt/test-driver :postgres
+    (testing "Can sync row count"
+      (mt/with-temp-test-data [["city"
+                                [{:field-name "name"
+                                  :base-type  :type/Text}]
+                                [["Los Angeles"]
+                                 ["Las Vegas"]]]]
+        (fn [conn]
+          ;; row count is estimated so we VACUUM so the statistic table is updated before syncing
+          (next.jdbc/execute! conn ["VACUUM;"])
```

Comment:
I'm a bit worried about this flaking, as there's no guarantee this fixes the statistics. AFAIK `ANALYZE` will be more reliable.

Reply:
> AFAIK ANALYZE will be more reliable.
interesting, how so?

on flaking: the test-data db is quite small so vacuuming shouldn't take a lot of time. and also no one modify it, so it should be pretty safe IOM.